assets/tests/components/propertiesPanel/vehiclePropertiesPanel.test.tsx

```diff
@@ -289,4 +291,51 @@ describe("VehiclePropertiesPanel", () => {
 
     expect(container.innerHTML).toContain("c-station-icon")
   })
+
+  test.each<{ tab: TabMode; clickTarget: string; initialTab?: TabMode }>([
+    { tab: "run", clickTarget: "Run" },
+    { tab: "block", clickTarget: "Block" },
+    { tab: "status", clickTarget: "Status", initialTab: "block" },
+  ])(
+    "when active tab changes to '$tab', fires fullstory event",
+    ({ tab, clickTarget, initialTab }) => {
+      mockFullStoryEvent()
+
+      render(
+        <VehiclePropertiesPanel
+          selectedVehicle={vehicleFactory.build()}
+          initialTab={initialTab}
+        />
+      )
+
+      fireEvent.click(screen.getByRole("tab", { name: clickTarget }))
+
+      expect(window.FS!.event).toHaveBeenCalledWith(
+        "Switched tab in Vehicle Properties Panel",
+        { tab_str: tab }
+      )
+    }
+  )
+
+  test.each<{ clickTarget: string; initialTab?: TabMode }>([
+    { clickTarget: "Run", initialTab: "run" },
+    { clickTarget: "Block", initialTab: "block" },
+    { clickTarget: "Status", initialTab: "status" },
+  ])(
+    "when active tab '$initialTab' is clicked, does not fire fullstory event",
+    ({ clickTarget, initialTab }) => {
+      mockFullStoryEvent()
+
+      render(
+        <VehiclePropertiesPanel
+          selectedVehicle={vehicleFactory.build()}
+          initialTab={initialTab}
+        />
+      )
+
+      fireEvent.click(screen.getByRole("tab", { name: clickTarget }))
```

Comment:
@mbta/skate-developers does anyone know when we should prefer `userEvent` vs `fireEvent`? 
IIRC I was getting test errors with `userEvent`, and thought that I was just using the wrong module, but upon reviewing, it seems that I've got them flipped.

I know we've had issues with `userEvent` and leaflet, but I don't think that's related here?

Reply:
Oops, this didn't get posted on the right LoC. 

I'm trying to refactor this over to `userEvent` now.

Reply:
I prefer `userEvent` unless there are specific issues with it like we've encountered with leaflet based on [the reasoning in the rtl common mistakes guide](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event)

Reply:
Speculation about the possible test errors you were encountering before that I missed on initial review - you may need to use `await` before calling `userEvent.click`

Reply:
> you may need to use await before calling userEvent.click

Quite possible that was the issue I first ran into!

Unfortunately I'm now hitting a error related to `XMLHttpRequest`, 
<details>
<summary>XMLHttpRequest Error</summary>

```
    Error: AggregateError
        at Object.dispatchError (/assets/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:63:19)
        at Request.<anonymous> (/assets/node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:655:18)
        at Request.emit (node:events:523:35)
        at ClientRequest.<anonymous> (/assets/node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:121:14)
        at ClientRequest.emit (node:events:511:28)
        at Socket.socketErrorListener (node:_http_client:495:9)
        at Socket.emit (node:events:511:28)
        at emitErrorNT (node:internal/streams/destroy:151:8)
        at emitErrorCloseNT (node:internal/streams/destroy:116:3)
        at processTicksAndRejections (node:internal/process/task_queues:82:21) {
      type: 'XMLHttpRequest'
    }
```

</details>

My current guess is that there is a component which doesn't have it's hooks mocked out.

Reply:
Just figured this out, seems that it was a combination of `useMinischedule{run,block}` and `useTripShape`.
See [`591eb31` (#2188)](https://github.com/mbta/skate/pull/2188/commits/591eb319d042ed82981062e995e47f17a6931b31)

Reply:
For future reference, the best way I found to debug this (after trying many things) was to spy on fetch and do `console.trace` within the fetch.